profiles/factory.py

```diff
@@ -1,10 +1,53 @@
-from flask import Flask
+from urllib.parse import urlencode
+
+from flask import Flask, jsonify, make_response, redirect, request
+from profiles.api.errors import OAuth2Error, ClientError
+from profiles.api.oauth2 import OAUTH2_BP
 from profiles.api.ping import PING_BP
+from profiles.utilities import remove_none_values
+from werkzeug.exceptions import HTTPException, InternalServerError
 
 
-def create_app():
+def create_app(config):
     app = Flask(__name__)
+    app.TRAP_HTTP_EXCEPTIONS = True
+    app.config.update(dict(
+        config=config
+    ))
 
+    app.register_blueprint(OAUTH2_BP, url_prefix='/oauth2')
     app.register_blueprint(PING_BP)
 
+    def http_error_handler(exception):
+        if isinstance(exception, ClientError):
+            return redirect(exception.uri + '?' + urlencode(remove_none_values({
+                'error': exception.error,
+                'error_description': exception.description,
+            })), exception.status_code)
+
+        if isinstance(exception, OAuth2Error):
+            body = remove_none_values({
+                'error': exception.error,
+                'error_description': exception.description,
+            })
+            return make_response(jsonify(body), exception.status_code)
+
+        if not isinstance(exception, HTTPException):
+            exception = InternalServerError(getattr(exception, 'message', None))
+
+        if request.path.startswith('/oauth2/authorize') or request.path.startswith('/oauth2/check'):
```

Comment:
an option we have in the future is to send the user to one of the error URL provided in the configuration. We have that, but it's probably not easy to access the right one from here